miasm2/expression/expression_dissector.py

```diff
@@ -0,0 +1,168 @@
+import miasm2.expression.expression as m2_expr
+
+
+class ExprDissector:
+    """
+    Dissects expressions into subexpressions
+    """
+
+    def __init__(self, registers=None, irdst=None):
+        """
+        Initialises the architecture variables
+        :param registers: registers of an architecture
+        :param irdst: IRA instruction pointer of an architecture
+        """
+        # set architecture variables
+        if registers and irdst:
+            self.arch_vars = set(registers + [irdst])
+
+    def _iterator(self, e):
+        """
+        Parses the arguments of an expression
+        :param e: expression
+        :return: iterator of a list of expressions
+        """
+        if isinstance(e, m2_expr.ExprOp):
+            iterator = iter(e.args)
+        elif isinstance(e, m2_expr.ExprSlice):
+            iterator = iter([e.arg])
+        elif isinstance(e, m2_expr.ExprCond):
+            iterator = iter((e.cond, e.src1, e.src2))
+        elif isinstance(e, m2_expr.ExprCompose):
+            iterator = iter((t[0] for t in e.args))
+        elif isinstance(e, m2_expr.ExprMem):
+            iterator = iter([e.arg])
+        elif isinstance(e, m2_expr.ExprAff):
+            iterator = iter([e.dst, e.src])
+        elif isinstance(e, m2_expr.ExprInt):
+            iterator = iter([])
+        elif isinstance(e, m2_expr.ExprId):
+            iterator = iter([])
+        else:
+            raise NotImplementedError()
+
+        return iterator
+
+    def _check_itetator(self, iterator, e):
+        """
+        Checks if an iterator is defined
+        :param iterator: iterator
+        :param e: expression
+        """
+        if not iterator:
+            raise NotImplementedError("no handler for {}".format((type(e))))
+
+    def _dissect(self, e, expr_type):
```

Comment:
When it is possible, please use at least two or three character for variable name, it is easier to read / search / avoid a mess when the function is more complex.